### ISSUE TITLE
Cannot find TestCase object on call stack

### ISSUE THREAD

**Issue by ro0NL** [sebastianbergmann/phpunit#5565]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 10.4.2
| PHP version         | 8.1.25
| Installation Method | PHAR

#### Summary

```
  PHPUnit 10.4.2 by Sebastian Bergmann and contributors.
  
  Runtime:       PHP 8.1.25
  Configuration: /app/phpunit.xml
  
  .EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE  63 / 205 ( 30%)
  EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE 126 / 205 ( 61%)
  EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEDD[17-Nov-2023 14:47:15 UTC] PHP Fatal error:  Allowed memory size of 3145728000 bytes exhausted (tried to allocate 266240 bytes) in /app/vendor/predis/predis/src/Connection/Replication/SentinelReplication.php on line 302
  
  Fatal error: Allowed memory size of 3145728000 bytes exhausted (tried to allocate 266240 bytes) in /app/vendor/predis/predis/src/Connection/Replication/SentinelReplication.php on line 302
  [17-Nov-2023 14:47:15 UTC] PHP Fatal error:  Uncaught PHPUnit\Event\Code\NoTestCaseObjectOnCallStackException: Cannot find TestCase object on call stack in phar:///usr/local/etc/tools/vendor/bin/phpunit.phar/phpunit/Event/Value/Test/TestMethodBuilder.php:51
  Stack trace:
  #0 phar:///usr/local/etc/tools/vendor/bin/phpunit.phar/phpunit/Runner/ErrorHandler.php(60): PHPUnit\Event\Code\TestMethodBuilder::fromCallStack()
  #1 [internal function]: PHPUnit\Runner\ErrorHandler->__invoke(2, 'Function must b...', '/app/vendor/sym...', 34)
  #2 /app/vendor/symfony/error-handler/Error/FatalError.php(34): xdebug_get_function_stack()
  #3 /app/vendor/symfony/error-handler/ErrorHandler.php(618): Symfony\Component\ErrorHandler\Error\FatalError->__construct('Error: Allowed ...', 0, Array, 2, false, NULL)
  #4 [internal function]: Symfony\Component\ErrorHandler\ErrorHandler::handleFatalError()
  #5 {main}
    thrown in phar:///usr/local/etc/tools/vendor/bin/phpunit.phar/phpunit/Event/Value/Test/TestMethodBuilder.php on line 51
  
  Fatal error: Uncaught PHPUnit\Event\Code\NoTestCaseObjectOnCallStackException: Cannot find TestCase object on call stack in phar:///usr/local/etc/tools/vendor/bin/phpunit.phar/phpunit/Event/Value/Test/TestMethodBuilder.php:51
  Stack trace:
  #0 phar:///usr/local/etc/tools/vendor/bin/phpunit.phar/phpunit/Runner/ErrorHandler.php(60): PHPUnit\Event\Code\TestMethodBuilder::fromCallStack()
  #1 [internal function]: PHPUnit\Runner\ErrorHandler->__invoke(2, 'Function must b...', '/app/vendor/sym...', 34)
  #2 /app/vendor/symfony/error-handler/Error/FatalError.php(34): xdebug_get_function_stack()
  #3 /app/vendor/symfony/error-handler/ErrorHandler.php(618): Symfony\Component\ErrorHandler\Error\FatalError->__construct('Error: Allowed ...', 0, Array, 2, false, NULL)
  #4 [internal function]: Symfony\Component\ErrorHandler\ErrorHandler::handleFatalError()
  #5 {main}
    thrown in phar:///usr/local/etc/tools/vendor/bin/phpunit.phar/phpunit/Event/Value/Test/TestMethodBuilder.php on line 51
```

#### How to reproduce

Im not sure yet. The offending PR on our side isnt that fancy, but i'll debug further Monday.

**Comment by ro0NL** [comment]
In symfony :eyes: 

```
I have no name!@aa7eeb552c8e:/app$ php -r "var_dump(xdebug_get_function_stack());"
[17-Nov-2023 18:13:30 UTC] PHP Warning:  Function must be enabled in php.ini by setting 'xdebug.mode' to 'develop' in Command line code on line 1

Warning: Function must be enabled in php.ini by setting 'xdebug.mode' to 'develop' in Command line code on line 1
array(0) {
}
```

**Comment by ro0NL** [comment]
This evolved into 

```
.EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE  63 / 205 ( 30%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEE 126 / 205 ( 61%)
EEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEEDDbin/test: line 44:    60 Segmentation fault      (core dumped) phpunit
```

Im trying to reason still where the segfault comes from

**Comment by sebastianbergmann** [comment]
Thank you for your report.

Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.

Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

**Comment by ro0NL** [comment]
I think the issue here is Symfony producing an unwanted warning from within their error-handler, causing phpunit's error-handler to trigger somewhat unexpectedly.

Im not sure it can be anticipated here better, as i find the whole custom error handling in PHP hard to follow at this point :)

I'll try to resolve the warning on Symfony's side first :+1: then will continue investigating the segfault, which occured after solving the original memory issues from the error output and likely triggered symfony's error handler in the first place.
